integration_tests/sdk/data_integration_tests/s3_test.py

```diff
@@ -0,0 +1,411 @@
+        artifact_type=ArtifactType.TABLE,
+        format="parquet",
+    )
+    assert dir_contents.type() == ArtifactType.TUPLE
+    assert dir_contents.get()[0].equals(hotel_reviews.get())
+
+    customers = data_integration.file(
+        "customers",
+        artifact_type=ArtifactType.TABLE,
+        format="parquet",
+    )
+    _save_artifact_and_check(
+        flow_manager,
+        data_integration,
+        artifact=customers,
+        format="parquet",
+        object_identifier="%s/%s" % (dir_name, customers_table_name),
+    )
+
+    # Check that both artifacts can be fetched by directory search.
+    dir_contents = data_integration.file(
+        dir_name + "/",
+        artifact_type=ArtifactType.TABLE,
+        format="parquet",
+    )
+
+    assert dir_contents.type() == ArtifactType.TUPLE
+    dir_contents_data = dir_contents.get()
+    assert len(dir_contents_data) == 2
+    assert dir_contents_data[0].equals(hotel_reviews.get()) and dir_contents_data[1].equals(
+        customers.get()
+    )
+
+
+def test_s3_fetch_directory_with_merge(flow_manager, data_integration):
+    dir_name = generate_object_name()
+
+    # Order hotel_reviews to be listed before customers by ordering the paths alphabetically.
+    hotel_reviews_table_name, customers_table_name = sorted(
+        [generate_table_name(), generate_table_name()]
+    )
+
+    # Write two tables into the directory.
+    hotel_reviews = data_integration.file(
+        "hotel_reviews", artifact_type=ArtifactType.TABLE, format="parquet"
+    )
+    _save_artifact_and_check(
+        flow_manager,
+        data_integration,
+        artifact=hotel_reviews,
+        format="parquet",
+        object_identifier="%s/%s" % (dir_name, hotel_reviews_table_name),
+    )
+
+    customers = data_integration.file(
+        "customers", artifact_type=ArtifactType.TABLE, format="parquet"
+    )
+    _save_artifact_and_check(
+        flow_manager,
+        data_integration,
+        artifact=customers,
+        format="parquet",
+        object_identifier="%s/%s" % (dir_name, customers_table_name),
+    )
+
+    expected_merged_data = pd.concat([hotel_reviews.get(), customers.get()], ignore_index=True)
+    dir_contents_merged = data_integration.file(
+        dir_name + "/",
+        artifact_type=ArtifactType.TABLE,
+        format="parquet",
+        merge=True,
+    )
+    assert dir_contents_merged.type() == ArtifactType.TABLE
+    assert expected_merged_data.equals(dir_contents_merged.get())
+
+
+def test_s3_non_tabular_fetch(client, flow_manager, data_integration):
+    string_data = "This is a string."
+    non_tabular_artifact = client.create_param("Non-Tabular Data", default=string_data)
+    assert non_tabular_artifact.type() == ArtifactType.STRING
+
+    _save_artifact_and_check(
+        flow_manager,
+        data_integration,
+        artifact=non_tabular_artifact,
+        format=None,
+    )
+
+
+def test_s3_fetch_multiple_files(client, flow_manager, data_integration):
```

Comment:
this LGTM. Will let @cw75 to double check

Reply:
This looks correct.